pkg/workflows/wasm/v2/writer.go

```diff
@@ -0,0 +1,19 @@
+package wasm
```

Comment:
nit: you could probably combine utils, writer and syncer files in this package into one, to reduce noise.

Reply:
copy/pasted and modified from V1. I forgot to rename the file.

Reply:
After the rename, the intent is clearer. X.go is a file that always compiles, X_test_hooks.go is meant to allow us to test, since we won't have the host, and X_wasip1 is the host callback(s).